src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -2469,6 +2469,63 @@ void Lowering::ContainCheckNeg(GenTreeOp* neg)
         }
     }
 }
+
+//----------------------------------------------------------------------------------------------
+// Try converting SELECT/SELECTCC to CINC/CINCCC. Conversion is possible only if
+// both the trueVal and falseVal are integral constants and abs(trueVal - falseVal) = 1.
+//
+// Arguments:
+//     select - The select node that is now SELECT or SELECTCC
+//     cond   - The condition node that SELECT or SELECTCC uses
+//
+void Lowering::TryLowerCselToCinc(GenTreeOp* select, GenTree* cond)
+{
+    assert(select->OperIs(GT_SELECT, GT_SELECTCC));
+
+    GenTree* trueVal  = select->gtOp1;
+    GenTree* falseVal = select->gtOp2;
+    if (trueVal->IsCnsIntOrI() && falseVal->IsCnsIntOrI() && cond->OperIsSimple() &&
```

Comment:
Can we flip this and have it as an early out instead?

Also, `OperIsSimple` should be `OperIsCompare()` and we can delay this check until we are actually going to reverse it below -- if we aren't going it reverse it then we can do this transformation for arbitrary conditions.

Reply:
Done